config/packages/messenger.yaml

```diff
@@ -8,19 +8,92 @@ framework:
             sync: "sync://"
             async:
                 dsn: "%env(MESSENGER_TRANSPORT_DSN)%"
+                options:
+                    queues:
+                        async: ~
+                    exchange:
+                        name: async
                 retry_strategy:
-                    max_retries: 5
+                    max_retries: 4
                     delay: 600000
                     multiplier: 10
```

Comment:
I propose the reduce the multiplier. delay is `600000` (=== 600 seconds). Multiplying this with a multiplayer of 10 means, 600000 ms, then 6000000 ms, then 60000000 ms.. and then 600000000 ms. Often you see a multiplier of a factor 2 or 3 in symfony messenger configurations. But not 10. 

600000000 is a delay of 7 days! This is too long IMO. I want to get rid of large queues faster, if it pills up, it will take days or weeks to get rid of. 

My proposal: reduce this multiplier down to 3 maybe 4 max.. Also decrease the `max_delay` option accordingly.

Reply:
I deliberately chose a week as the max delay, because: servers have down time and sometimes they have a lot of downtime. A longer max delay provides better stability for the fediverse (yeah we are not a big part of it, but so what). I was fine to remove the fifth retry, because trying for 2 weeks is a bit much, I agree. I am also fine with trying more often so if the server comes back up after a day we do not need to carry the message around for another 6 days. 
So my proposal would be to decrease the max delay and increase the retry amount, so that in the end we do try for 7 days, once per day or something like that

Reply:
I still might not resolve the issue when a messenger never will be successfully processed, no matter if the server is back online. 

OK, I like the direction so we at least do it faster to potentially empty the queue faster. That is a good start, I would go for max delay of 5 days (432000000). Decrease the normal delay to 5min and like you mentioned increase the retries to 6x. After some calculations, I propose:

```yaml
                retry_strategy:
                    max_retries: 6
                    delay: 300000
                    multiplier: 4
                    max_delay: 432000000
```

Explanation:

- 300000 ms = 5 min.
- Max reties of 6 (so 6 times *  delay * multiplier)
- And multiplier of 4. Meaning: 5 min, 20 min, ~1 hour, ~5 hours, 1 day, 3.5 days with a max delay of 5 days.

Reply:
```yaml
                retry_strategy:
                    max_retries: 8
                    delay: 400000
                    multiplier: 6
                    max_delay: 86400000
```
This way we have these delays:
1. ~7 minutes
2. 40 minutes
3. 4 hours
4. 24 hours
5. 24 hours
6. 24 hours
7. 24 hours
8. 24 hours

So we have an even number as a cap which is exactly 24 hours and we retry for a total of 5 days  4 hours and ~47 minutes. The more frequent retry means that messages can quit the queue faster however in step 4-8 they will be in the same delay queue because they are all delayed by the same amount in that step